File: c#/program50.cs
using System;

class StringX
{
    public string str;

    public StringX(string s)
    {
        this.str = s;
    }

    public string StringReverse()
    {
        int start = 0,end = 0;
        char temp = '\0';

        char[] Arr = str.ToCharArray();

        start = 0;
        end = str.Length-1;

        while(start < end)
        {
            temp = Arr[start];
            Arr[start] = Arr[end];
            Arr[end] = temp;

            start++;
            end--;
        }

        string s = new string(Arr);
        return s;
    }
}

class MainClass
{
    static void Main()
    {
        string str = "",sRet = "";

        Console.WriteLine("Enter string :");
        str = Console.ReadLine();

        StringX obj = new StringX(str);
        sRet = obj.StringReverse();

        Console.WriteLine("Reverse string is : "+sRet);
    }
}

File: java/program50.java
import java.util.*;

class StringX
{
    public String str;

    StringX(String s)
    {
        this.str = s;
    }

    public String StringReverse()
    {
        char Arr[] = str.toCharArray();    
        int start = 0;
        int end = Arr.length-1;
        char temp = ' ';

        while(start < end)
        {
            temp = Arr[start];
            Arr[start] = Arr[end];
            Arr[end] = temp;

            start++;
            end--;
        }

        String name = new String(Arr);
        return name;
    }
}

class program50
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string :");
        String str = sobj.nextLine();

        StringX obj = new StringX(str);
        
        System.out.println("Reverse String is : "+obj.StringReverse());
    }
}